### DISCUSSION TITLE
Internal error when manually running the job

### DISCUSSION THREAD

**Discussion by gbetous-swir** [community/community#25730] [Actions]
<p>Hi,</p>
<p>I created a public repository dedicated to use GitHub action with all I need for my real CI.</p>
<p>I have some basic jobs (simple script that echo “hello” and “goodbye” to a name), and I want to be able to give a list of names (in real CI it will be a list of supported targets, matrix usage) and select which greeting I want (in real CI it will be sanity and nightly tests).</p>
<p>I think I have a working skeleton, but for some reason, the manual tests are not working at all : I’m facing a “GitHub Actions has encountered an internal error when running your job.” error message.</p>
<p>Any idea ?</p>
<p>Thanks a lot,</p>
<p>The repo : <a href="https://github.com/gbetous-swir/GHActions-demo" class="inline-onebox" rel="noopener nofollow ugc">GitHub - gbetous-swir/GHActions-demo: Demo of GitHub actions</a><br>
The error message : <a href="https://github.com/gbetous-swir/GHActions-demo/actions/runs/1713024643" class="inline-onebox" rel="noopener nofollow ugc">Initial commit · gbetous-swir/GHActions-demo@a573fc5 · GitHub</a></p>

**Comment by ChristopherHX** [comment, marked as answer]
<p>Hi,</p>
<p>You have been hit by <a href="https://github.com/actions/runner/issues/1483" class="inline-onebox" rel="noopener nofollow ugc">Boolean inputs are not actually booleans · Issue #1483 · actions/runner · GitHub</a>, which ends up assigning string ‘true’ / ‘false’ to a boolean workflow_call parameter =&gt; internal server error.</p>
<p>How to fix your workflow?<br>
Cast your workflow_call boolean inputs to actual boolean values.</p>
<pre><code class="lang-yaml">manual_test:
    name: manual test
    uses: gbetous-swir/GHActions-demo/.github/workflows/jobs.yml@main
    with:
       usernames:  ${{ github.event.inputs.names }}
       hello_job:  ${{ github.event.inputs.hello == 'true' }}
       goodbye_job:  ${{ github.event.inputs.goodbye == 'true' }}
</code></pre>
<p>or</p>
<pre><code class="lang-yaml">manual_test:
    name: manual test
    uses: gbetous-swir/GHActions-demo/.github/workflows/jobs.yml@main
    with:
       usernames:  ${{ github.event.inputs.names }}
       hello_job:  ${{ fromjson(github.event.inputs.hello) }}
       goodbye_job:  ${{ fromjson(github.event.inputs.goodbye) }}
</code></pre>
<p><s>I have no idea how GitHub has blown up their error messages, but I’m shure they had better error messages while reusable workflows were in beta.</s></p>
<p><em>I sometimes feel like beeing an alpha tester of <a href="http://github.com" rel="noopener nofollow ugc">github.com</a> free for ghes, since reusable workflows beta was more stable than what we test now.</em></p>

**Comment by gbetous-swir** [comment]
<p>Thank you very much, this was my issue, it works now :+1:</p>
<aside class="onebox githubactions" data-onebox-src="https://github.com/gbetous-swir/GHActions-demo/actions/runs/1716982997">
  <header class="source">

      <a href="https://github.com/gbetous-swir/GHActions-demo/actions/runs/1716982997" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>

  <article class="onebox-body">
    <div class="github-row">
  <div class="github-icon-container" title="Commit">
      <svg width="24" height="24" class="github-icon github-icon-success" viewbox="0 0 24 24" aria-hidden="true"><path d="M1 12C1 5.925 5.925 1 12 1s11 4.925 11 11-4.925 11-11 11S1 18.075 1 12zm16.28-2.72a.75.75 0 00-1.06-1.06l-5.97 5.97-2.47-2.47a.75.75 0 00-1.06 1.06l3 3a.75.75 0 001.06 0l6.5-6.5z"></path></svg>



  </div>

  <div class="github-info-container">
    <h4>
      <a href="https://github.com/gbetous-swir/GHActions-demo/actions/runs/1716982997" target="_blank" rel="noopener nofollow ugc">FIX booelan issue on manual tests</a>
    </h4>

      <div class="github-info">
        Manual run <span class="github-run-number">#5</span>
      </div>
  </div>
</div>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>
